src/main/java/io/vertx/core/DeploymentOptions.java

```diff
@@ -111,23 +120,43 @@ public DeploymentOptions setConfig(JsonObject config) {
     return this;
   }
 
+  /**
+   * How the verticle(s) should be deployed?
+   *
+   * @return the verticle thread mode
+   */
+  public ThreadMode getThreadMode() {
+    return threadMode;
+  }
+
+  /**
+   * Set how the verticle(s) should be deployed, e.g. a worker or a virtual thread verticle
+   *
+   * @param threadMode the thread mode
+   * @return a reference to this, so the API can be used fluently
+   */
+  public DeploymentOptions setThreadMode(ThreadMode threadMode) {
+    this.threadMode = threadMode;
+    return this;
+  }
+
   /**
    * Should the verticle(s) be deployed as a worker verticle?
    *
-   * @return true if will be deployed as worker, false otherwise
+   * @return {@code true} if will be deployed as worker, {@code false} otherwise
    */
   public boolean isWorker() {
-    return worker;
+    return threadMode == ThreadMode.WORKER;
   }
 
   /**
    * Set whether the verticle(s) should be deployed as a worker verticle
    *
-   * @param worker true for worker, false otherwise
+   * @param worker {@code true} for worker, {@code false} force event-loop
    * @return a reference to this, so the API can be used fluently
    */
   public DeploymentOptions setWorker(boolean worker) {
-    this.worker = worker;
+    this.threadMode = worker ? ThreadMode.WORKER : ThreadMode.EVENT_LOOP;
```

Comment:
`ThreadingModel.XYZ` would read better than `ThreadMode.XYZ`, but maybe that's just me 😉

Reply:
good idea